nifi-extension-bundles/nifi-parquet-bundle/nifi-parquet-content-viewer/src/main/java/org/apache/parquet/web/controller/ParquetContentViewerController.java

```diff
@@ -0,0 +1,155 @@
+        } catch (final ResourceNotFoundException e) {
+            logger.warn("Content not found", e);
+            response.sendError(HttpURLConnection.HTTP_NOT_FOUND, "Content not found");
+            return;
+        } catch (final AccessDeniedException e) {
+            logger.warn("Content access denied", e);
+            response.sendError(HttpURLConnection.HTTP_FORBIDDEN, "Content access denied");
+            return;
+        } catch (final Exception e) {
+            logger.warn("Content retrieval failed", e);
+            response.sendError(HttpURLConnection.HTTP_INTERNAL_ERROR, "Content retrieval failed");
+            return;
+        }
+
+        response.setStatus(HttpServletResponse.SC_OK);
+
+        final boolean formatted = Boolean.parseBoolean(request.getParameter("formatted"));
+        if (!formatted) {
+            final InputStream contentStream = downloadableContent.getContent();
+            contentStream.transferTo(response.getOutputStream());
+            return;
+        }
+
+        // allow the user to drive the data type but fall back to the content type if necessary
+        String displayName = request.getParameter("mimeTypeDisplayName");
+        if (displayName == null) {
+            displayName = downloadableContent.getType();
+        }
+
+        if (displayName == null || !(displayName.equals("parquet") || displayName.equals("application/vnd.apache.parquet"))) {
+            response.sendError(HttpURLConnection.HTTP_BAD_REQUEST, "Unknown content type");
+            return;
+        }
+
+        try {
+            //Convert InputStream to a seekable InputStream
+            byte[] data = getInputStreamBytes(downloadableContent.getContent());
+
+            if (data.length == 0) {
+                response.getOutputStream().write("Content size is too large to display.".getBytes());
+                return;
+            }
+
+            Configuration conf = new Configuration();
+
+            //Allow older deprecated schemas
+            conf.setBoolean("parquet.avro.readInt96AsFixed", true);
+
+            final InputFile inputFile = new NifiParquetInputFile(new ByteArrayInputStream(data), data.length);
+            final ParquetReader<GenericRecord> reader = AvroParquetReader.<GenericRecord>builder(inputFile)
+                    .withConf(conf)
+                    .build();
+
+            //Format and write out each record
+            GenericRecord record;
+            objectMapper.getFactory().configure(JsonGenerator.Feature.AUTO_CLOSE_TARGET, false);
+            boolean firstRecord = true;
+            while ((record = reader.read()) != null) {
+                if (firstRecord) {
+                    firstRecord = false;
+                } else {
+                    response.getOutputStream().write(",\n".getBytes());
+                }
+
+                objectMapper.writerWithDefaultPrettyPrinter().writeValue(response.getOutputStream(), objectMapper.readTree(record.toString()));
+            }
```

Comment:
It would be helpful to write valid JSON. In this case, writing an opening `[` and a closing `]`, with a `,` after each record would produce a valid JSON array.